rampwf/score_types/detection.py

```diff
@@ -310,15 +374,20 @@ def _select_minipatch_tuples(y_list, minipatch):
     """
     row_min, row_max, col_min, col_max = minipatch
 
-    y_list = np.asarray(y_list)
+    y_list = np.asarray(y_list).T
 
-    y_list_cut = ((y_list[0] >= col_min) & (y_list[0] < col_max) &
-                  (y_list[1] >= row_min) & (y_list[1] < row_max))
+    y_list_cut = ((y_list[0] >= row_min) & (y_list[0] < row_max) &
+                  (y_list[1] >= col_min) & (y_list[1] < col_max))
 
     return y_list_cut
 
 
-def _match_tuples(y_true, y_pred, minipatch=None):
+def filter_minipatch_tuples(y_list, minipatch):
+    in_minipatch = _select_minipatch_tuples(y_list, minipatch)
+    return np.array(y_list)[in_minipatch].tolist()
```

Comment:
this could also be:

```
 [obj for (obj, cond) in zip(y_list, in_minipatch) if cond]
```

which preserves the list of tuples, and is for small lists (which is always the case since we are inside a single patch) also much faster (no numpy overhead)